numpy/core/src/multiarray/scalarapi.c

```diff
@@ -25,7 +25,13 @@ _descr_from_subtype(PyObject *type)
     PyObject *mro;
     mro = ((PyTypeObject *)type)->tp_mro;
     if (PyTuple_GET_SIZE(mro) < 2) {
-        return PyArray_DescrFromType(NPY_OBJECT);
+        PyErr_SetString(PyExc_TypeError,
+                "Could not get dtype for NumPy scalar subclass, this can "
+                "happen depending on how it is inherited and whether a dtype "
+                "was registered.  In general, subclassing scalars is not "
+                "encouraged. Please get in touch with the NumPy developers "
+                "for more information.");
```

Comment:
Yes, its not a nice error message, but I honestly don't know who should or can run into this...

Reply:
can you provide an example script of when this function is called ? i see it getting called from PyArray_DescrFromTypeObject but wasnt able to come up with a use case.

Reply:
Honestly, I don't think its possible right now due to python ensuring a strict `mro` for C defined subclasses...

This error _will_ probably be reachable in some stranger C-API calls when someone subclasses from `np.generic` but not a concrete dtype.  I.e. for `class mydt(np.generic): pass`  we currently do guess the `dtype`, but there is no actual dtype defined.

Maybe I should just include this into the other few changes to split that up a bit clearer, since I really do not think it can be reached currently.